docs/how-to-guides/others/an-example-procedure-for-adding-a-new-node.md

```diff
@@ -0,0 +1,89 @@
+
+In this context, AWSIM is presumed.
+Therefore, [AWSIM simulator](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/digital-twin-simulation/awsim-tutorial/) can be useful.
+If you are using actual hardware, please refer to the [How-to guides](https://autowarefoundation.github.io/autoware-documentation/main/how-to-guides/).
+
+## 2. Recording a rosbag using Autoware
+
+Before developing a new node, it is recommended to record a rosbag in order to evaluate.
+If you need a new sensor, you should add it to your vehicle or AWSIM.
+
+In this case, it is recommended to save all topics regardless of whether they are necessary or not.
+For example, in Localization, since the initial position estimation service is triggered by the input to rviz and the GNSS topic, the initial position estimation does not start when playing back data unless those topics are saved.
+
+Consider the use of the [mcap format](https://mcap.dev/) if data capacity becomes a concern.
+
+It is worth noting that using `ros2 bag record` increases computational load and might affect performance.
+After data recording, verifying the smooth flow of sensor data and unchanged time series is advised.
+This verification can be accomplished, for example, by inspecting the image data with `rqt_image_view` during `ros2 bag play`.
+
+## 3. Developing the new node
+
+When developing a new node, it could be beneficial to reference a package that is similar to the one you intend to create.
+
+It is advisable to thoroughly read the [Design page](https://autowarefoundation.github.io/autoware-documentation/main/design/), contemplate the addition or replacement of nodes in Autoware, and then implement your solution.
+
+For example, a node doing NDT, a LiDAR-based localization method, is [ndt_scan_matcher](https://github.com/autowarefoundation/autoware.universe/tree/main/localization/ndt_scan_matcher).
+If you want to replace this with a different approach, implement a node which produces the same topics and provides the same services.
+
+`ndt_scan_matcher` is launched as [pose_estimator](https://github.com/autowarefoundation/autoware.universe/blob/main/launch/tier4_localization_launch/launch/pose_estimator/pose_estimator.launch.xml), so it is necessary to replace the launch file as well.
+
+## 4. Evaluating by a rosbag-based simulator
+
+Once the new node is implemented, it is time to evaluate it.
+[logging_simulator](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/rosbag-replay-simulation/) is a tool of how to evaluate the new node using the rosbag captured in step 2.
+
+It works by replacing the command to start Autoware with logging_simulator.
+
+For example, if the AWSIM-compatible execution command is
+
+```bash
+ros2 launch autoware_launch e2e_simulator.launch.xml ...
+```
+
+then the command to start logging_simulator is
+
+```bash
+ros2 launch autoware_launch logging_simulator.launch.xml ...
+```
+
```

Comment:
I think this explanation is difficult to understand and not important. So you may remove it.
Instead, how about adding the following?

```suggestion
When you run the logging_simulator, you can set planning:=false or control:=false to disable the launch of specific component nodes.
`ros2 launch autoware_launch logging_simulator.launch.xml ... planning:=false control:=false`

```